codepipeline/codepipeline.tf

```diff
@@ -0,0 +1,228 @@
+resource "aws_codepipeline" "default" {
+  name     = "${var.name_prefix}-codepipeline"
+  role_arn = aws_iam_role.codepipeline_role.arn
+
+  artifact_store {
+    location = aws_s3_bucket.default.bucket
+    type     = "S3"
+  }
+
+  stage {
+    name = "Source"
+
+    action {
+      category = "Source"
+      name     = "Source"
+      owner    = "AWS"
+      provider = "CodeStarSourceConnection"
+      version  = "1"
+
+      output_artifacts = ["source_output"]
+
+      configuration = {
+        ConnectionArn    = aws_codestarconnections_connection.default.arn
+        FullRepositoryId = var.repo_name
+        BranchName       = var.repo_branch_name
+      }
+
+    }
+
+  }
+
+  stage {
+    name = "Build"
+
+    action {
+      name             = "Build"
+      category         = "Build"
+      owner            = "AWS"
+      provider         = "CodeBuild"
+      input_artifacts  = ["source_output"]
+      output_artifacts = ["build_output"]
+      version          = "1"
+
+      configuration = {
+        ProjectName = aws_codebuild_project.default.name
+      }
+    }
+  }
+}
+
+resource "aws_codebuild_project" "default" {
+  name         = "${var.name_prefix}-codebuild"
+  service_role = aws_iam_role.codebuild_role.arn
+
+  artifacts {
+    type = "CODEPIPELINE"
+  }
+
+  environment {
+    compute_type                = "BUILD_GENERAL1_SMALL"
+    image                       = "aws/codebuild/amazonlinux2-x86_64-standard:2.0"
+    type                        = "LINUX_CONTAINER"
+    image_pull_credentials_type = "CODEBUILD"
+  }
+
+  logs_config {
+    cloudwatch_logs {
+      status = "ENABLED"
+    }
+  }
+
+  source {
+    type = "CODEPIPELINE"
+  }
+}
+
+resource "aws_s3_bucket" "default" {
+  bucket = "${var.name_prefix}-bucket"
```

Comment:
You will probably want to use [bucket_prefix](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket#bucket_prefix) instead of bucket to reduce the chance of collisions.

Reply:
Done.